File: angular-gpp/src/app/models/UsuarioModel.ts
import { DepartamentoModel } from './departamento.model';
import { PefllUsuarioModel } from './pefll-usuario.model';

export class UsuarioModel {
  id: number;
  uid: number;
  nome: string;
  email: string;
  idFilial: number;
  accessToken: string;
  tokenType: string;
  emailVerifiedAt: string;
  password: string;
  rememberToken: string;
  createdAt: string;
  updatedAt: string;
  active: boolean;
  iddepto: string;
  foto: string;
  departamento: DepartamentoModel;
  petfllUsuario: PefllUsuarioModel;

  constructor(
    id: number,
    uid: number,
    nome: string,
    email: string,
    idFilial: number,
    accessToken: string,
    tokenType: string,
    emailVerifiedAt: string,
    password: string,
    rememberToken: string,
    createdAt: string,
    updatedAt: string,
    active: boolean,
    iddepto: string,
    foto: string,
    departamento: DepartamentoModel,
    petfllUsuario: PefllUsuarioModel
  ) {
    this.id = id;
    this.uid = uid;
    this.nome = nome;
    this.email = email;
    this.idFilial = idFilial;
    this.accessToken = accessToken;
    this.tokenType = tokenType;
    this.emailVerifiedAt = emailVerifiedAt;
    this.password = password;
    this.rememberToken = rememberToken;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.active = active;
    this.iddepto = iddepto;
    this.foto = foto;
    this.departamento = departamento;
    this.petfllUsuario = petfllUsuario;
  }
}

File: gpp-api/src/main/java/com/develop/gpp/domain/entity/UsuarioModel.java
package com.develop.gpp.domain.entity;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "usuario")
public class UsuarioModel {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long uid;

    private String nome;

    private String email;

    @Column(name = "idfilial")
    private Long idFilial;

    private String accessToken;

    private String tokenType;

    @Column(name = "email_verified_at")
    private LocalDateTime emailVerifiedAt;

    private String password;

    @Column(name = "remember_token")
    private String rememberToken;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    private Boolean active;

    private String iddepto;

    private String foto;

    @ManyToOne
    @JoinColumn(name = "iddepto", referencedColumnName = "iddepto", insertable = false, updatable = false)
    private DepartamentoModel departamento;

    // @OneToOne(mappedBy = "usuario")
    // private PefllUsuarioModel petfllUsuario;



    // Getters and Setters
    // ...
}
